### PR TITLE
Styling details and error handling on profile pages

### LINKED ISSUE #117: Handle 5xx errors on profile pages
If something goes wrong with the API call, the profile page does not inform the user

### PR DESCRIPTION
**Pull request by tiagojsag**
Fixes #117 

- When loading data on profile search page, it now shows a message on failure
- On the same page, the CSS is no longer broken while data is loading
- On profile pages (actor and place) an error message is displayed in case data fails to load

### CODE REVIEW

File: frontend/scripts/pages/profile-actor.page.js [hunk 1]
```diff
@@ -276,12 +276,15 @@ const _setEventListeners = () => {
...
 
-const _showErrorMessage = () => {
+const _showErrorMessage = (message = null) => {
   const el = document.querySelector('.l-profile-actor');
   el.classList.add('-error');
   document.querySelector('.js-loading').classList.add('is-hidden');
   el.querySelector('.js-wrap').classList.add('is-hidden');
   el.querySelector('.js-error-message').classList.remove('is-hidden');
+  if (message !== null) {
+    el.querySelector('.message').innerHTML = message;
```
Comment (sorodrigo) [inline frontend/scripts/pages/profile-actor.page.js]: Its a good practice to add js specific classes for DOM operations i.e. `.js-message`.

File: frontend/scripts/pages/profile-place.page.js [hunk 2]
```diff
@@ -178,12 +178,16 @@ const _setEventListeners = () => {
...
 
-const _showErrorMessage = () => {
+const _showErrorMessage = (message = null) => {
   const el = document.querySelector('.l-profile-place');
   document.querySelector('.js-loading').classList.add('is-hidden');
   el.classList.add('-error');
   el.querySelector('.js-wrap').classList.add('is-hidden');
   el.querySelector('.js-error-message').classList.remove('is-hidden');
+  if (message !== null) {
+    el.querySelector('.message').innerHTML = message;
```
Comment (sorodrigo) [inline frontend/scripts/pages/profile-place.page.js]: same

File: frontend/styles/layouts/l-profiles.scss [hunk 3]
```diff
@@ -41,6 +53,20 @@
         font-size: $font-size-medium;
       }
 
+      > .c-search > .js-search-input {
```
Comment (sorodrigo) [inline frontend/styles/layouts/l-profiles.scss]: Avoid using `js` prefixed classes for styling.

File: frontend/scripts/pages/profile-actor.page.js [hunk 4]
```diff
@@ -325,15 +328,12 @@ const _init = ()  => {
...
-        return null;
-      }
-
-      if (response.status === 200) {
+      if (response.ok) {
         return response.json();
       }
-    })
+      _showErrorMessage(response.statusText);
+    }, (reason) => _showErrorMessage(reason.message)
```
Comment (sorodrigo) [inline frontend/scripts/pages/profile-actor.page.js]: Nifty af! I didn't know of the existence of Promise.then second argument. 🙌 
As a side note, I prefer using `.catch` for handling errors but this is perfectly fine.

File: frontend/scripts/pages/profile-actor.page.js [hunk 5]
```diff
@@ -325,15 +328,12 @@ const _init = ()  => {
...
+    )
     .then((result) => {
       if (!result) return;
```
Comment (sorodrigo) [inline frontend/scripts/pages/profile-actor.page.js]: On a second thought, if we use catch, we don't have to early return in the next then.

**Comment by tiagojsag** [comment]
I think i got them all. let me know @sorodrigo

**Pull request merged**